### PR TITLE
add build system support for parmetis 

### PR DESCRIPTION
**Pull request by cyrush**

### CODE REVIEW

**Comment by cyrush** [comment]
@xjrc the smoke test should provide a good starting point for us, can you take a look and see if it makes sense to you?

File: src/tests/thirdparty/t_parmetis_smoke.cpp [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+        std::cout << std::endl;
+    }
+
+    // Finalize MPI
+    MPI_Finalize();
+
+    if(res == METIS_OK)
+        return 0;
+    else // (res == METIS_ERROR)
+        return -1;
```
Comment (xjrc) [inline src/tests/thirdparty/t_parmetis_smoke.cpp]:
```suggestion
        return (res == METIS_OK) ? 0 : -1;
```

File: src/tests/thirdparty/t_parmetis_smoke.cpp [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+
+    /*
+    The eptr and eind arrays are similar in nature to the xadj and adjncy
+    arrays used to specify the adjacency list of a graph but now for each
+    element they specify the set of nodes that make up each element.
+    Specifically, the set of nodes that belong to element i is stored in
+    array eind starting at index eptr[i] and ending at (but not including)
+    index eptr[i + 1] (in other words, eind[eptr[i]] up through and including
+    eind[eptr[i + 1]-1]). Hence, the node lists for each element are stored
+    consecutively in the array eind.
```
Comment (xjrc) [inline src/tests/thirdparty/t_parmetis_smoke.cpp:49]: It looks like METIS uses the same offsetting mechanism as some of our client codes; interesting!

**Review by xjrc — APPROVED** [review]
Looks great, Cyrus; thanks for getting this in!

File: src/config/ConduitConfig.cmake.in [hunk 3]
```diff
@@ -34,6 +34,7 @@ if(NOT CONDUIT_FOUND)
     set(CONDUIT_HDF5_DIR  "@HDF5_DIR@")
     set(CONDUIT_ADIOS_DIR "@ADIOS_DIR@")
     set(CONDUIT_SILO_DIR "@SILO_DIR@")
+    set(CONDUIT_PARMETIS_DIR "@PARMETIS_DIR@")
     set(CONDUIT_PYTHON_ENABLED "@PYTHON_FOUND@")
```
Comment (cyrush) [inline src/config/ConduitConfig.cmake.in:39]: we also need to add METIS_DIR here

File: src/config/conduit_config.mk.in [hunk 4]
```diff
@@ -28,6 +28,8 @@ CONDUIT_SILO_DIR  = @SILO_DIR@
 CONDUIT_ADIOS_DIR = @ADIOS_DIR@
 CONDUIT_ZFP_DIR   = @ZFP_DIR@
 
+CONDUIT_PARMETIS_DIR = @PARMETIS_DIR@
```
Comment (cyrush) [inline src/config/conduit_config.mk.in:32]: also need to add METIS_DIR

**Pull request merged**